app/com/linkedin/drelephant/tunin/AutoTuningAPIHelper.java

```diff
@@ -0,0 +1,387 @@
+    tuningJobExecution.isDefaultExecution = tuningJobExecutionDefault.isDefaultExecution;
+    tuningJobExecution.tuningAlgorithm = tuningJobExecutionDefault.tuningAlgorithm;
+    tuningJobExecution.paramSetState = ParamSetStatus.CREATED;
+    tuningJobExecution.save();
+
+    logger.debug("New Default tuning execution: " + Json.toJson(tuningJobExecution));
+
+    List<JobSuggestedParamValue> jobSuggestedParamValueList =
+        JobSuggestedParamValue.find
+            .where()
+            .eq(JobSuggestedParamValue.TABLE.jobExecution + "." + JobExecution.TABLE.id,
+                tuningJobExecutionDefault.jobExecution.id).findList();
+
+    //Save default parameters corresponding to new default execution
+    for (JobSuggestedParamValue jobSuggestedParamValue : jobSuggestedParamValueList) {
+      JobSuggestedParamValue jobSuggestedParamValue1 = new JobSuggestedParamValue();
+      jobSuggestedParamValue1.id = 0;
+      jobSuggestedParamValue1.jobExecution = jobExecution;
+      jobSuggestedParamValue1.paramValue = jobSuggestedParamValue.paramValue;
+      jobSuggestedParamValue1.tuningParameter = jobSuggestedParamValue.tuningParameter;
+      jobSuggestedParamValue1.save();
+    }
+
+    tuningJobExecution =
+        TuningJobExecution.find
+            .select("*")
+            .where()
+            .eq(TuningJobExecution.TABLE.jobExecution + "." + JobExecution.TABLE.id, tuningJobExecution.jobExecution.id)
+            .setMaxRows(1).findUnique();
+
+    return tuningJobExecution;
+  }
+
+  /**
+   * Set default values for allowed max execution time and resource usage in case not provided in API call
+   * And set tuning algorithm based on value of job type and optimization metric
+   * @param tuningInput
+   */
+  public void setDefaultValue(TuningInput tuningInput) {
+    Configuration configuration = ElephantContext.instance().getAutoTuningConf();
+    if (tuningInput.getAllowedMaxExecutionTimePercent() == null) {
+      Double allowedMaxExecutionTimePercent = new Double(Utils.getNonNegativeInt(configuration, "", 150));
+      tuningInput.setAllowedMaxExecutionTimePercent(allowedMaxExecutionTimePercent);
+    }
+    if (tuningInput.getAllowedMaxResourceUsagePercent() == null) {
+      Double allowedMaxResourceUsagePercent = new Double(Utils.getNonNegativeInt(configuration, "", 150));
+      tuningInput.setAllowedMaxResourceUsagePercent(allowedMaxResourceUsagePercent);
+    }
+
+    TuningAlgorithm tuningAlgorithm =
+        TuningAlgorithm.find.select("*").where().eq(TuningAlgorithm.TABLE.jobType, tuningInput.getJobType())
+            .eq(TuningAlgorithm.TABLE.optimizationMetric, tuningInput.getOptimizationMetric()).findUnique();
+    tuningInput.setTuningAlgorithm(tuningAlgorithm);
+  }
+
+  /**
+   * Handles the api request and returns param suggestions as response
+   * @param tuningInput Rest api parameters
+   * @return Parameter Suggestion
+   */
+  public Map<String, Double> getCurrentRunParameters(TuningInput tuningInput) {
```

Comment:
I recall this method name being used in Application.java. Can you clarify the distinction? Please update the method name accordingly to avoid confusion.

Reply:
This class is the Helper class for the API call. So same name.